web/pgadmin/tools/sqleditor/static/js/components/sections/Query.jsx

```diff
@@ -291,9 +291,15 @@ export default function Query({onTextSelect, handleEndOfLineChange}) {
       }
     });
 
-    eventBus.registerListener(QUERY_TOOL_EVENTS.CHANGE_EOL, (lineSep)=>{
+    eventBus.registerListener(QUERY_TOOL_EVENTS.CHANGE_EOL, (lineSep, queryChanged)=>{
+      // Set the new EOL character in the editor.
       editor.current?.setEOL(lineSep);
-      eventBus.fireEvent(QUERY_TOOL_EVENTS.QUERY_CHANGED, true);
+      // Enable/disable save button state if changes are ongoing, otherwise mark editor as clean.
+      if (queryChanged){
```

Comment:
there is no need to check queryChanged. When the first the file is loaded, you can detect the EOL, set it and mark the doc as clean. After that, whenever EOL changes, isDirty will take care. Here query tool is deciding whether mark the doc clean or dirty, which is anti-pattern.

Reply:
Ohk now got your point.